### PR TITLE
dma: prevent bitshift ub

### PR DESCRIPTION
**Pull request by slawblauciak**
Signed-off-by: Slawomir Blauciak <slawomir.blauciak@linux.intel.com>

### CODE REVIEW

File: src/drivers/intel/cavs/hda-dma.c [hunk 1]
```diff
@@ -229,21 +229,20 @@ static void hda_dma_get_dbg_vals(struct hda_chan_data *chan,
...
 			uint32_t wp = \
-				(dbg_data->last_wp[HDA_DBG_PRE] << 16) | \
-				(dbg_data->last_wp[HDA_DBG_POST] & 0xFFFF); \
+				((uint32_t)dbg_data->last_wp[HDA_DBG_PRE] << \
+				 16) | dbg_data->last_wp[HDA_DBG_POST]; \
 			uint32_t rp = \
-				(dbg_data->last_rp[HDA_DBG_PRE] << 16) | \
-				(dbg_data->last_rp[HDA_DBG_POST] & 0xFFFF); \
+				((uint32_t)dbg_data->last_rp[HDA_DBG_PRE] << \
+				 16) | dbg_data->last_rp[HDA_DBG_POST]; \
```
Comment (lgirdwood) [inline src/drivers/intel/cavs/hda-dma.c]: Can we have some macros for these magic numbers to match the comments explaining the data format ?

Comment (slawblauciak) [reply src/drivers/intel/cavs/hda-dma.c]: I've decided to create generic macros for such operations

File: src/include/sof/sof.h [hunk 2]
```diff
@@ -54,6 +54,14 @@ struct sa;
 	__attribute__((unused))		\
 	typedef char META_CONCAT(assertion_failed_, MESSAGE)[(COND) ? 1 : -1]
 
+/* merges two 16-bit values into a single 32-bit value */
+#define merge_16b16b(high, low) (((uint32_t)(high) << 16) | \
```
Comment (lgirdwood) [inline src/include/sof/sof.h]: Good idea, can you put these in math/numbers.h as that's where there is similar code. Thanks

Comment (slawblauciak) [reply src/include/sof/sof.h]: Done

**Pull request merged**